libstuff/libstuff.cpp

```diff
@@ -1948,6 +1975,34 @@ string SAESDecrypt(const string& buffer, const string& ivStr, const string& key)
     return SAESDecrypt(buffer, iv, key);
 }
 
+// These decrypt functions are used to return a value that still includes possible
+// padding, so it is up the caller to manage stripping the potential NULL chars off the end.
+string SAESDecryptNoStrip(const string& buffer, const size_t& bufferSize, unsigned char* iv, const string& key) {
```

Comment:
Similar to above - can we have the stripping one call this one and then do the stripping?